00-basics/003-tests/Udemi/Fetch and Promises/script.js

```diff
@@ -0,0 +1,78 @@
+
+const body = document.querySelector('body');
+const input = document.querySelector('input');
+//////////////////////////////////////
+const div = document.createElement('div');
+// 1. Fetch and Promises PVZ:
+// const getData = function (country) {
+//   fetch(`https://restcountries.eu/rest/v2/name/${country}`) //Return Promise
+//     .then(response => {
+//       return response.json();
+//     })
+//     .then(results => {
+//       console.log(results);
+//     })
+//     .catch(err => {
+//       console.log(err);
+//     });
+// };
+//
+// getData('lithuania');
+//////////////////////
+
+// 2. Print Country:
+// const getData = function (country) {
+//     fetch(`https://restcountries.eu/rest/v2/name/${country}`) //Return Promise
+//         .then(response => {
+//             // 404 err
+//             if(!response.ok){
+//                 throw new Error( `Country not found (${response.status})`)
+//             }
+//             return response.json();
+//         })
+//         .then(results => {
+//             console.log(results);
+//             div.innerHTML = ` <div>
+//     <img class="country__img"  src="${results[0].flag}"/>
+//     <div class="country__data">
+//       <h3 class="country__name">${results[0].name}</h3>
+//       <h4 class="country__region">${results[0].region}</h4>
+//     </div>
+//   </div>
+//     `;
+//         })
+//         .catch(err => {
+//             console.log(err);
+//            div.innerHTML = `Error (${err})`; // 404 err in HTML
+//         });
+//     body.appendChild(div);
+// };
+//
+// getData('lithuania');
+// // getData('germany');
+////////////////////
+
+// Search from countries by name:
+
+input.addEventListener('keyup', function(e){
```

Comment:
TBH, that feels like a very complicated way to get a promise to play with.
You can easily get a promise in Node.js by simply doing `return Promise.resolve(someVariableValue)`, or if you want to simulate something more realistic, just perform an fs-promise operation on any file: https://nodejs.org/dist/latest/docs/api/fs.html#fs_promises_api